filament/src/main/java/net/fabricmc/filament/task/mappingio/CheckMappingsTask.java

```diff
@@ -1,62 +1,136 @@
 package net.fabricmc.filament.task.mappingio;
 
+import java.io.File;
 import java.io.IOException;
 import java.nio.file.Path;
 import java.util.ArrayList;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
 
+import net.fabricmc.mappingio.MappedElementKind;
+import net.fabricmc.mappingio.MappingVisitor;
+
+import org.gradle.api.provider.Property;
+import org.gradle.api.tasks.Input;
 import org.gradle.api.tasks.TaskAction;
+import org.jetbrains.annotations.Nullable;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import net.fabricmc.filament.task.base.FilamentTask;
-import net.fabricmc.filament.task.base.WithFileInput;
 import net.fabricmc.mappingio.MappingReader;
-import net.fabricmc.mappingio.tree.MappingTree;
-import net.fabricmc.mappingio.tree.MemoryMappingTree;
 
-public abstract class CheckMappingsTask extends FilamentTask implements WithFileInput {
+public abstract class CheckMappingsTask extends FilamentTask {
 	private static final Logger LOGGER = LoggerFactory.getLogger(CheckMappingsTask.class);
 
+	@Input
+	abstract Property<File> getMappings();
```

Comment:
Should be a `DirectoryProperty` with the `@InputDirectory` annotation.

Reply:
Fixed!